### PR TITLE
Update docs to avoid Shpinx errors and warnings

### LINKED ISSUE #291: Documentation does not compile properly
Here's the problem-reporting output of "sudo make install":

   sphinx-build -c ./src/build-data/sphinx -b html ./doc/manual build/docs/manual
   Running Sphinx v1.3.1
   WARNING: the config value 'latex_show_urls' has type `bool', defaults to `str.'
   loading pickled environment... not yet created
   building [mo]: targets for 0 po files that are out of date
   building [html]: targets for 26 source files that are out of date
   updating environment: 26 added, 0 changed, 0 removed
   reading sources... [100%] x509                                                                                
   /Users/ur20980/Src/botan/doc/manual/tls.rst:197: WARNING: Error when parsing function declaration:
   If no return type {
   Invalid definition: Expected identifier in nested name. [error at 353]
     TLS::Client( output_fn output, data_cb data, alert_cb alert, handshake_cb handshake_complete,    TLS::Session_Manager& session_manager, Credentials_Manager& credendials_manager, const    TLS::Policy& policy, RandomNumberGenerator& rng, const Server_Information& server_info, const    Protocol_Version offer_version, const std::vector<std::string>& app_protocols,
     ---------------------------------------------------------------------------------------------------------------------------------------   -------------------------------------------------------------------------------------------------------------------------------------------   -------------------------------------------------------------------------------^
   } else if return type {
   If declId, parameters, and qualifiers {
   Invalid definition: Expected identifier in nested name. [error at 11]
     TLS::Client( output_fn output, data_cb data, alert_cb alert, handshake_cb handshake_complete,    TLS::Session_Manager& session_manager, Credentials_Manager& credendials_manager, const   TLS::Policy& policy, RandomNumberGenerator& rng, const Server_Information& server_info, const    Protocol_Version offer_version, const std::vector<std::string>& app_protocols,
     -----------^
   } else If parenthesis in noptr-declarator {
   Invalid definition: Expecting "(" in parameters_and_qualifiers. [error at 23]
     TLS::Client( output_fn output, data_cb data, alert_cb alert, handshake_cb handshake_complete,    TLS::Session_Manager& session_manager, Credentials_Manager& credendials_manager, const    TLS::Policy& policy, RandomNumberGenerator& rng, const Server_Information& server_info, const    Protocol_Version offer_version, const std::vector<std::string>& app_protocols,
     -----------------------^
   }
   }
   /Users/ur20980/Src/botan/doc/manual/tls.rst:197: WARNING: Error when parsing function declaration:
   If no return type {
   Invalid definition: Expecting "(" in parameters_and_qualifiers. [error at 7]
  size_t reserved_io_buffer_size)
  -------^
   } else if return type {
   Invalid definition: Expecting "(" in parameters_and_qualifiers. [error at 30]
     size_t reserved_io_buffer_size)
     ------------------------------^
   } 
   /Users/ur20980/Src/botan/doc/manual/tls.rst:287: WARNING: Error when parsing function declaration:
   If no return type {
   Invalid definition: Expected identifier in nested name. [error at 362]
     TLS::Server( std::function<void, const byte*, size_t> output_fn, std::function<void, const byte*, size_t>    data_cb, std::function<TLS::Alert, const byte*, size_t> alert_cb, TLS::Session_Manager&    session_manager, Credentials_Manager& creds, const TLS::Policy& policy, RandomNumberGenerator&    rng, std::function<std::string, std::vector<std::string> > proto_chooser,
     ---------------------------------------------------------------------------------------------------------------------------------------    -------------------------------------------------------------------------------------------------------------------------------------------  ----------------------------------------------------------------------------------------^
   } else if return type {
   If declId, parameters, and qualifiers {
   Invalid definition: Expected identifier in nested name. [error at 11]
  TLS::Server( std::function<void, const byte*, size_t> output_fn, std::function<void, const byte*, size_t> data_cb, std::function<TLS::Alert, const byte*, size_t> alert_cb, TLS::Session_Manager& session_manager, Credentials_Manager& creds, const TLS::Policy& policy, RandomNumberGenerator& rng, std::function<std::string, std::vector<std::string> > proto_chooser,
  -----------^
} else If parenthesis in noptr-declarator {
Invalid definition: Expecting "(" in parameters_and_qualifiers. [error at 54]
  TLS::Server( std::function<void, const byte*, size_t> output_fn, std::function<void, const byte*, size_t> data_cb, std::function<TLS::Alert, const byte*, size_t> alert_cb, TLS::Session_Manager& session_manager, Credentials_Manager& creds, const TLS::Policy& policy, RandomNumberGenerator& rng, std::function<std::string, std::vector<std::string> > proto_chooser,
  ------------------------------------------------------^
}
}
/Users/ur20980/Src/botan/doc/manual/tls.rst:287: WARNING: Error when parsing function declaration:
If no return type {
Invalid definition: Expected identifier in nested name, got keyword: const [error at 5]
  const std::vector<std::string>& protocols, bool is_datagram = false, bool reserved_io_buffer_size)
  -----^
} else if return type {
Invalid definition: Expecting "(" in parameters_and_qualifiers. [error at 41]
  const std::vector<std::string>& protocols, bool is_datagram = false, bool reserved_io_buffer_size)
  -----------------------------------------^
}
/Users/ur20980/Src/botan/doc/manual/tls.rst:3: SEVERE: Duplicate ID: "TLS::Session".
/Users/ur20980/Src/botan/doc/manual/tls.rst:745: WARNING: Type must be either just a name or a typedef-like declaration.
Just a name error: Invalid definition: Expected identifier in nested name, got keyword: enum [error at 4]
  enum Version_Code
  ----^
Typedef-like expression error: Invalid definition: Expected identifier in nested name, got keyword: enum [error at 4]
  enum Version_Code
  ----^
/Users/ur20980/Src/botan/doc/manual/tls.rst:3: SEVERE: Duplicate ID: "TLS::Protocol_Version::latest_tls_version".
/Users/ur20980/Src/botan/doc/manual/tls.rst:3: SEVERE: Duplicate ID: "TLS::Protocol_Version::latest_dtls_version".
/Users/ur20980/Src/botan/doc/manual/x509.rst:5: SEVERE: Duplicate ID: "X509_Certificate::subject_dnC".
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] x509                                                                                 
generating indices... genindex py-modindex
writing additional pages... search
copying static files... done
copying extra files... done
dumping search index in English (code: en) ... done
dumping object inventory... done
build succeeded, 10 warnings.
./src/scripts/install.py --destdir=/opt/local --build-dir="build" --bindir=bin --libdir=lib --docdir=doc --includedir=include
   INFO: Botan 1.11.21 installation complete
```

### PR DESCRIPTION
**Pull request by webmaster128**
Closes #291

### CODE REVIEW

File: doc/manual/tls.rst [hunk 1]
```diff
@@ -742,20 +744,10 @@ The ``TLS::Protocol_Version`` class represents a specific version:
...
- .. cpp:function:: static Protocol_Version latest_tls_version()
-
-      Returns the latest version of TLS supported by this implementation
-      (currently TLS v1.2)
-
- .. cpp:function:: static Protocol_Version latest_dtls_version()
-
-      Returns the latest version of DTLS supported by this implementation
-      (currently DTLS v1.2)
-
```
Comment (randombit) [inline doc/manual/tls.rst]: Why remove these?

Comment (randombit) [inline doc/manual/tls.rst]: Nevermind now I see the problem

File: doc/manual/x509.rst [hunk 2]
```diff
@@ -24,10 +24,6 @@ in the :doc:`tls` protocol. A X.509 certificate is represented by
 
      Returns the distinguished name of the issuer
 
- .. cpp:function:: X509_DN subject_dn() const
```
Comment (randombit) [inline doc/manual/x509.rst]: This isn't redundant exactly, this second one should be for `issuer_dn` (and then `subject_dn` should be corrected re what name it returns)

Comment (webmaster128) [inline doc/manual/x509.rst]: See line 23–25

Comment (webmaster128) [inline doc/manual/x509.rst]: Ah, sorry. I misunderstood your comment. I'll fix that.

**Comment by webmaster128** [comment]
@randombit Could you have a second look at 41a7025?

**Pull request merged**